cmd/modelcmd/credentials.go

```diff
@@ -104,15 +91,40 @@ func GetCredentials(
 		},
 	)
 	if err != nil {
-		return nil, "", "", errors.Annotatef(
-			err, "finalizing %q credential for cloud %q",
-			credentialName, args.Cloud.Name,
-		)
+		return nil, "", "", AnnotateWithFinalizationError(err, credentialName, args.Cloud.Name)
 	}
 
 	return credential, credentialName, regionName, nil
 }
 
+// FinalizeFileContent replaces the path content of cloud credentials "file" attribute with its content.
+func FinalizeFileContent(credential *cloud.Credential, provider environs.EnvironProvider) (*cloud.Credential, error) {
```

Comment:
This function has been factored out of the existing code. Part of the reason for this bug is there is limited testing around this particular area of code. No test verifies that the file path is actually is actually converted to its content. Three things come to my mind here, we can either inject the `readFile` function into all relevant functions so that we can "mock" the file reading for test. Alternatively, we could use a file I/O mocking library. Lastly, we can make an actual temp file for testing. I am aware of the fundamental Pros and Cons of each method. 

Another thing to do might be to relocate `FinalizeFileContent`  so that would live in `Cloud` where it might be added to the interface and stubbed the method altogether.  I welcome any feedback regarding this issue.